connector/src/main/java/org/geysermc/connector/network/translators/chat/MessageTranslator.java

```diff
@@ -75,30 +70,43 @@
      * @param locale Locale to use for translation strings
      * @return Parsed and formatted message for bedrock
      */
-    public static String convertMessage(String message, String locale) {
-        Component component = GsonComponentSerializer.gson().deserialize(message);
+    public static String convertMessage(Component message, String locale) {
+        try {
+            // Get a Locale from the given locale string
+            Locale localeCode = Locale.forLanguageTag(locale.replace('_', '-'));
```

Comment:
Not super relevant to this PR - I wonder if we should optimize this, so we don't have to call `replace` every time. Relevant StackOverflow: https://stackoverflow.com/questions/16228992/commons-lang-stringutils-replace-performance-vs-string-replace

Reply:
Not sure we can as I believe all the locales we receive from the client and use for filenames use `_` whereas the built-in Java locales use `-`. But I can optimise it to use StringUtils, granted we have the lib installed.

Reply:
This would mean we have to include the apache commons-lang lib, so we decided against it.